### ISSUE TITLE
sfdisk refuses to create small partition at the end of disk

### ISSUE THREAD

**Issue by wferi** [karelzak/util-linux#1249]
With util-linux 2.36.1-7 (Debian sid) the following succeeds:
```
# dd if=/dev/zero of=boot.img bs=1M count=11
11+0 records in
11+0 records out
11534336 bytes (12 MB, 11 MiB) copied, 0.00983671 s, 1.2 GB/s
sfdisk boot.img <<EOF
> 2048,16384,83,*
18432,,8e,-
EOF
Checking that no-one is using this disk right now ... OK

Disk boot.img: 11 MiB, 11534336 bytes, 22528 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes

>>> Created a new DOS disklabel with disk identifier 0x6e96b730.
boot.img1: Created a new partition 1 of type 'Linux' and of size 8 MiB.
boot.img2: Created a new partition 2 of type 'Linux LVM' and of size 2 MiB.
boot.img3: Done.

New situation:
Disklabel type: dos
Disk identifier: 0x6e96b730

Device     Boot Start   End Sectors Size Id Type
boot.img1  *     2048 18431   16384   8M 83 Linux
boot.img2       18432 22527    4096   2M 8e Linux LVM

The partition table has been altered.
Syncing disks.
```
Cool. So the second partition is 4096 sectors (2 MiB) as expected. However, if I start with an image that is 1 MiB smaller:
```
# dd if=/dev/zero of=boot.img bs=1M count=10
10+0 records in
10+0 records out
10485760 bytes (10 MB, 10 MiB) copied, 0.00683336 s, 1.5 GB/s
# /sbin/sfdisk boot.img <<EOF
2048,16384,83,*
18432,,8e,-
EOF
Checking that no-one is using this disk right now ... OK

Disk boot.img: 10 MiB, 10485760 bytes, 20480 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes

>>> Created a new DOS disklabel with disk identifier 0x5fbd2455.
boot.img1: Created a new partition 1 of type 'Linux' and of size 8 MiB.
boot.img2: All space for primary partitions is in use.
Failed to add #2 partition: Invalid argument
Leaving.
```
Bummer.
I expected it to create a 1 MiB `#2` partition, so I consider this a bug.
I guess it's related to the cylinder granularity stemming from the artificial geometry (255 heads and 63 sectors/track, which can't be overridden since 2.26), but since the partitions aren't cylinder-aligned in the 11 MiB case either, I don't understand this limitation.

**Comment by karelzak** [comment]
This is not related to cylinder granularity as the code does not CHS stuff at all.

Unfortunately, this is related to the end-of-partition calculation when checking minimal partition size against the size of the device ... classic one-by-one error :-(

**Comment by karelzak** [comment]
The problem will be fixed in v2.37 and v2.36.2. Thanks for your report.

**Comment by wferi** [comment]
> This is not related to cylinder granularity as the code does not use CHS stuff at all.

I'm glad to hear that. :)

> Unfortunately, this is related to the end-of-partition calculation when checking minimal partition size against the size of the device ... classic one-by-one error :-(

Many thanks for the quick fix!
